### ISSUE TITLE
Installation fails when Rprofile prints to stdout

### ISSUE THREAD

**Issue by brookslogan** [amices/mice#646]
**Before submitting the issue**
- Check whether a bug report is the right format for your issue. For questions about using the `mice` package, see [Discussions](https://github.com/amices/mice/discussions). For theoretical background about the `mice` methods, see [Flexible Imputation of Missing Data](https://stefvanbuuren.name/fimd/).
- Please make sure that you are using the latest version of `mice` (e.g. with the `packageVersion()` or `sessionInfo()` function).
- Please verify whether the bug has already been addressed (e.g. on [GitHub](https://github.com/amices/mice/issues?q=is%3Aissue) or [Stack Overflow](https://stackoverflow.com/questions/tagged/r-mice).

**Describe the bug**

When a home Rprofile exists, and that .Rprofile produces output to stdout, installation of mice fails.  (Likely the same for site Rprofiles without home Rprofiles.)

**To Reproduce**
Create a **repr**oducible **ex**ample, see e.g. the [`reprex` package](https://reprex.tidyverse.org/). A good reprex uses simulated or public data to showcase the bug. We recommend using one of the incomplete datasets included in the `mice` package (e.g. `mice::boys`) and the `reprex()` function from the `reprex` package.

Paste the code and output of your reprex here.

``` r
home_rprofile_path <- "~/.Rprofile" # assuming no R_PROFILE setting
if (file.exists(home_rprofile_path )) {
  stop ('Home .Rprofile already exists; refusing to clobber with this test.')
  # (`file(home_rprofile_path, "x")` would be preferable but isn't supported)
} else { # `stop` doesn't actually stop in `reprex`
  withr::with_file(home_rprofile_path, {
    writeLines('cat("Using settings from ~/.Rprofile\n")', home_rprofile_path)
    if ("mice" %in% installed.packages()) {
      remove.packages("mice")
    }
    install.packages("mice")
  })
}
#> Installing package into '/home/fullname/R/x86_64-redhat-linux-gnu-library/4.1'
#> (as 'lib' is unspecified)
#> Warning in install.packages("mice"): installation of package 'mice' had
#> non-zero exit status
```

<sup>Created on 2024-06-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Running within `reprex` appears to reproduce the problem but not the error message to the console; this is the relevant part from running outside reprex:
```
g++ -m64 -std=gnu++14 -shared -L/usr/lib64/R/lib -Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -Wl,--build-id=sha1 -o mice.so RcppExports.o legendre.o match.o matchindex.o Using settings from ~/.Rprofile -L/usr/lib64/R/lib -lR
/usr/bin/ld: cannot find Using: No such file or directory
/usr/bin/ld: cannot find settings: No such file or directory
/usr/bin/ld: cannot find from: No such file or directory
/usr/bin/ld: cannot find ~/.Rprofile: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [/usr/share/R/make/shlib.mk:10: mice.so] Error 1
ERROR: compilation failed for package ‘mice’
```

**Expected behavior**
A clear and concise description of what you expect to happen without the bug.

Installation succeeds.

**Screenshots**

**Additional context**

This appears to occur due to [this line](https://github.com/amices/mice/blob/99bd724eac99ae44a3b7e8b2b5da0f2ddd264ec6/src/Makevars#L2C13-L2C56) (or [this Windows equivalent](https://github.com/amices/mice/blob/99bd724eac99ae44a3b7e8b2b5da0f2ddd264ec6/src/Makevars.win#L3)).  Note that `Rcpp:::LdFlags()` is [deprecated and produces an empty string](https://www.rdocumentation.org/packages/Rcpp/versions/1.0.12/topics/LdFlags-deprecated), and [hasn't been needed for long enough](https://github.com/RcppCore/Rcpp/issues/1302#issuecomment-2082670237) that it likely could be dropped now.

User workarounds include temporarily renaming the problematic Rprofile, or producing output to stderr (e.g., with `message()`) instead of stdout in the Rprofile.

**Comment by stefvanbuuren** [comment]
Before building, have you tried removing all `src/*.o` and `src/*.so` files?

**Comment by brookslogan** [comment]
I didn't have a `src/` directory (within my working directory at least), as I was attempting to install from CRAN or RSPM.  Here's some testing of devtools-based installation:
- Cloning and installing or re-installing with `devtools::install('~/<path>/mice', quick = TRUE, build = FALSE, upgrade = FALSE, keep_source = TRUE)` (from ESS) works even with a problematic home Rprofile.  Existing `src/*.o` files don't seem to interfere with re-installation, and `devtools::clean_dll()`ing also doesn't introduce problems. 
- Attempting to install directly with `devtools::install_github("amices/mice")` has the same error as `install.packages()` though.
- (`pak::pkg_install()` appears to succeed in building and installing directly from CRAN/RSPM from source, even after clearing its cache.)

I should note that I am on Fedora, R version 4.1.3 (2022-03-10), devtools_2.4.5, Rcpp_1.0.12.
- If Fedora has a different configuration than other systems (e.g., where it builds packages or maybe the lack of an R_BUILD_ENVIRON setting?), that might factor in.
- If R and/or devtools have updated the way builds are done and attempt to isolate them more in later versions by ignoring the home and maybe site Rprofiles, then this may no longer be a problem.

**Comment by stefvanbuuren** [comment]
Thanks for trying. 

I am not on Fedora and cannot reproduce. I fear I won't be able to help.

**Comment by brookslogan** [comment]
Thanks for checking on this.  The fix for me was as simple as #647, though I haven't done extensive testing.  Thought I'd at least throw it in a PR as an option.
